grafana-plugin/src/pages/incidents/Incidents.tsx

```diff
@@ -719,13 +737,28 @@ class Incidents extends React.Component<IncidentsPageProps, IncidentsPageState>
 
   clearPollingInterval() {
     clearInterval(this.pollingIntervalId);
-    this.pollingIntervalId = undefined;
+    this.pollingIntervalId = null;
   }
 
   setPollingInterval(filters: IncidentsFiltersType = this.state.filters, isOnMount = false) {
-    this.pollingIntervalId = setInterval(() => {
-      this.fetchIncidentData(filters, isOnMount);
-    }, POLLING_NUM_SECONDS * 1000);
+    const startPolling = (delayed = false) => {
+      this.pollingIntervalId = setTimeout(
+        async () => {
+          const isBrowserWindowInactive = document.hidden;
```

Comment:
nice idea! but are you stopping/starting polling on `document.hidden` change?

Reply:
I could have done that but I just didn't want to add more logic (subscribe to event, unsubscribe etc) in the class. If the tab is inactive, it won't call the backend, if it becomes active again it will detect that in the `setTimeout`.